socialhome/tests/test_viewsets.py

```diff
@@ -25,3 +25,41 @@ def test_image_upload(self):
         self.assertIn("http://127.0.0.1:8000/media/uploads/", url)
         image_upload = ImageUpload.objects.filter(user=self.user).first()
         self.assertEqual(image_upload.image.name, url.replace("http://127.0.0.1:8000/media/", ""))
+
+
+class TestObtainSocialhomeAuthToken(SocialhomeAPITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.user = UserFactory()
+
+    def test_user_missing(self):
+        response = self.client.post(reverse("api-token-auth"))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.data["username"], ["This field is required."])
+
+    def test_password_missing(self):
+        response = self.client.post(reverse("api-token-auth"))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.data["password"], ["This field is required."])
+
+    def test_unable_to_login(self):
+        response = self.client.post(reverse("api-token-auth"),
+                                    {"username": self.user.username, "password": "%sLol" % self.user.password})
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.data, {'non_field_errors': ['Unable to log in with provided credentials.']})
+
+    def test_authentication(self):
+        self.client.force_authenticate(self.user, token="0ab231f9f7ab63a6f78fd9746b9bdc9926a734fd")
+        response = self.client.post(reverse("api-token-auth"),
```

Comment:
Since this is a `SocialhomeAPITestCase` you don't need to use `self.client`. Our test classes use some handy helper stuff which means you can just do `self.post("api-token-auth", {"username": self.user.username, "password": self.user.password})`. (see [here](https://django-test-plus.readthedocs.io/en/latest/usage.html#testing-drf-views))

Does that work? If not I'm not sure what could be the problem. The password should be set by the factory so that shouldn't be an issue..